libc/utils/MPFRWrapper/MPFRUtils.cpp

```diff
@@ -974,6 +1007,22 @@ template bool compare_unary_operation_single_output<float16>(Operation, float16,
                                                              RoundingMode);
 #endif
 
+template <typename OutType, typename InType>
+bool compare_unary_narrower_operation_single_output(Operation op, InType input,
```

Comment:
Maybe change `compare_unary_operation_single_output` to 2 template parameters instead.